test/terraform_ssh_example_test.go

```diff
@@ -91,6 +137,164 @@ func configureTerraformOptions(t *testing.T, exampleFolder string) (*terraform.O
 	return terraformOptions, keyPair
 }
 
+func testScpDirFromHost(t *testing.T, terraformOptions *terraform.Options, keyPair *aws.Ec2Keypair) {
+	// Run `terraform output` to get the value of an output variable
+	publicInstanceIP := terraform.Output(t, terraformOptions, "public_instance_ip")
+
+	// We're going to try to SSH to the instance IP, using the Key Pair we created earlier, and the user "ubuntu",
+	// as we know the Instance is running an Ubuntu AMI that has such a user
+	publicHost := ssh.Host{
+		Hostname:    publicInstanceIP,
+		SshKeyPair:  keyPair.KeyPair,
+		SshUserName: "ubuntu",
+	}
+
+	// It can take a minute or so for the Instance to boot up, so retry a few times
+	maxRetries := 30
+	timeBetweenRetries := 5 * time.Second
+	description := fmt.Sprintf("SSH to public host %s", publicInstanceIP)
+	remoteTempFolder := "/tmp/testFolder"
+	remoteTempFilePath := fmt.Sprintf("%s/test.foo", remoteTempFolder)
+	remoteTempFilePath2 := fmt.Sprintf("%s/test.baz", remoteTempFolder)
```

Comment:
Always use `filepath.Join`. Building paths through string concatenation almost always leads to a bug.

Reply:
fixed